extensions/table.c

```diff
@@ -155,13 +219,49 @@ static table_row *row_from_string(cmark_syntax_extension *self,
       node_cell *cell = (node_cell *)parser->mem->calloc(1, sizeof(*cell));
       cell->buf = cell_buf;
       cell->start_offset = offset;
-      cell->end_offset = offset + cell_matched - 1;
+      if (cell_matched > 0)
+        cell->end_offset = offset + cell_matched - 1;
+      else
+        cell->end_offset = offset;
 
       while (cell->start_offset > 0 && string[cell->start_offset - 1] != '|') {
         --cell->start_offset;
         ++cell->internal_offset;
       }
 
+      cell->cell_data = (node_cell_data *)parser->mem->calloc(1, sizeof(node_cell_data));
+
+      // Check for a column-spanning cell
+      if (row->n_columns > 0 && cmark_strbuf_len(cell->buf) == 0 && cell->start_offset == cell->end_offset) {
+        cell->cell_data->colspan = 0;
+
+        // find the last cell that isn't part of a colspan, and increment that colspan
+        cmark_llist *tmp = row->cells;
+        node_cell *colspan_cell = NULL;
+        while (tmp) {
```

Comment:
Is there no way to iterate backwards on the row cells? Also should we raise some sort of error if the row is leading with a colspan marker?

Reply:
The cells are stored as a `cmark_llist`, which is a one-way linked list, so i can only scan forward, not backward.